src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -681,6 +685,38 @@ public TaskResource updateNotes(String taskId, NotesRequest notesRequest) {
         return cftTaskDatabaseService.findByIdOnly(taskId);
     }
 
+    /**
+     * Retrieve task role information.
+     * This method retrieves role permission information for a given task.
+     * The task should have a read permission to retrieve role permission information.
+     *
+     * @param taskId        the task id.
+     * @param accessControlResponse the access control response containing user id and role assignments.
+     * @return collection of roles
+     */
+    public List<TaskRolePermissions> getTaskRolePermissions(String taskId,
+                                                            AccessControlResponse accessControlResponse) {
+        final Optional<TaskResource> taskResource = getTaskById(taskId);
+        if (taskResource.isEmpty()) {
+            throw new TaskNotFoundException(TASK_NOT_FOUND_ERROR);
+        }
+        final Specification<TaskResource> taskResourceSpecification = TaskResourceSpecification
+            .buildTaskRolePermissionsQuery(taskResource.get().getTaskId(), accessControlResponse);
+
+        final Optional<TaskResource> taskResourceQueryResult = cftTaskDatabaseService.findTaskBySpecification(
+            taskResourceSpecification);
+
+        if (taskResourceQueryResult.isEmpty()) {
+            return emptyList();
+        }
```

Comment:
This is a bit different to the query endpoint.  If the user does not have the correct permissions to read the Task then we should probably return a 403 saying they don't have the permissions.  An empty list implies that there are no permissions linked to the Task.

Reply:
The above scenario is covered in RWA-810. I have written based on AC.

Reply:
I think this should be addressed in this PR since it is incorrect as it stands and therefore should not be merging it into master like that.
If the scenario is covered in another PR in its entirety then you can just remove this if statement and address it in the error scenario ticket.

Reply:
Removed from this PR.

Reply:
The specification returns optional element, so I need to check if its present before accessing it.

Reply:
If it is empty, throwing 403 exception

Reply:
Excellent.  I can see a test for it too.